### DISCUSSION TITLE
Include ids in headers when building the plugin pages

### DISCUSSION THREAD

**Discussion by Andre601** [woodpecker-ci/woodpecker#2882] [Request a Feature]
Would be useful if the parser used for parsing the Docs pages of the Woodpecker plugins could include header IDs.

That way could the maintainers of these doc pages link to a header section.

What I mean is, that when the `docs.md` is parsed should the tool used turn this:
```markdown
## Settings

something something...
```
Into this:
```html
<h2 id="settings">Settings</h2>
<p>something something...</p>
```
Also, if doable, maybe include a link with icon for a permanent link?

**Comment by anbraten** [comment]
I guess that should be possible by setting some option for the markdown parser we use: https://github.com/woodpecker-ci/woodpecker/blob/main/docs/plugins/woodpecker-plugins/src/markdown.ts#L26

Maybe you can have a look and adjust it.

**Reply by anbraten** [reply]
see https://www.npmjs.com/package/marked-gfm-heading-id
